### PR TITLE
Support for new functions run_at, entity_exists, extented support for function get_state

### PR DESCRIPTION
**Pull request by foxcris**
Hi,

first of all thanks for this great project. It helped me a lot. For my project https://github.com/foxcris/appdaemon-blinds-control some functionality was missing so i added the required functionality. 

Here what i have done:
- added support for function run_at
- added support for function entity_exists
- added support for calling get_state without using any entity_id 
(request states for all enities)
- added testcases for all changes

I hope i meet all the requirements for a contribution. If you have questions or if something is missing or should be reworked please let me know.

### CODE REVIEW

**Comment by FlorianKempenich** [comment]
Fantastic, thanks a lot for your comments and your contribution 😊

I'll look at it in a couple of days 👍

File: test/test_state.py [hunk 1]
```diff
@@ -60,9 +64,18 @@ def test_set_and_get_all_attribute(given_that, automation: MockAutomation):
...
+    assert automation.get_complete_state_dictionary() == {COVER: {'attributes': {'current_position': 0,
+                                                                                                'friendly_name': COVER},
+                                                                                'state': 'closed'},
+                                                          LIGHT: {'attributes': {'brightness': 11, 'color': 'blue'},
+                                                                                'state': 'on'}}
 
 @pytest.mark.only
 def test_throw_typeerror_when_attributes_arg_not_passed_via_keyword(given_that, automation: MockAutomation):
     given_that.state_of(LIGHT).is_set_to('on', attributes={'brightness': 11, 'color': 'blue'})
+    given_that.state_of(COVER).is_set_to('closed', attributes={'brightness': 11, 'color': 'blue'})
```
Comment (FlorianKempenich) [inline test/test_state.py]: This is unnecessary. Could you please remove it?

File: test/test_state.py [hunk 2]
```diff
@@ -60,9 +64,18 @@ def test_set_and_get_all_attribute(given_that, automation: MockAutomation):
     given_that.state_of(LIGHT).is_set_to('on', attributes={'brightness': 11, 'color': 'blue'})
     assert automation.get_all_attributes_from_light() == {'brightness': 11, 'color': 'blue'}
 
+def test_get_complete_state_dictionary(given_that, automation: MockAutomation):
```
Comment (FlorianKempenich) [inline test/test_state.py]: Good job adding a test 👍

File: test/test_miscellaneous_helper_functions.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+    def get_entity_exists(self, entity):
+        return self.entity_exists(entity)
+
+@automation_fixture(WithMiscellaneousHelperFunctions)
+def with_miscellaneous_helper_functions():
+    pass
+
+
+def test_entity_exitsts_true(given_that, with_miscellaneous_helper_functions, hass_functions):
```
Comment (FlorianKempenich) [inline test/test_miscellaneous_helper_functions.py]: Small typo in the test name: 'exitsts' -> 'exists'

File: test/test_miscellaneous_helper_functions.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    given_that.state_of(COVER).is_set_to("closed", {'friendly_name': f"{COVER}", 'current_position': 0})
+    assert with_miscellaneous_helper_functions.get_entity_exists(COVER)==True
+
+def test_entity_exitsts_false(given_that, with_miscellaneous_helper_functions, hass_functions):
```
Comment (FlorianKempenich) [inline test/test_miscellaneous_helper_functions.py]: Small typo in the test name: 'exitsts' -> 'exists'

File: test/test_state.py (hunk 1 above)
Comment (foxcris) [reply test/test_state.py]: You are right, line removed.

File: test/test_miscellaneous_helper_functions.py (hunk 3 above)
Comment (foxcris) [reply test/test_miscellaneous_helper_functions.py]: Done.

File: test/test_miscellaneous_helper_functions.py (hunk 4 above)
Comment (foxcris) [reply test/test_miscellaneous_helper_functions.py]: Done.

**Comment by foxcris** [comment]
I reworked my code according to your comments and pushed it to my repo.

**Comment by FlorianKempenich** [comment]
Great! 😃

**Comment by FlorianKempenich** [comment]
And . . . merged 😁
Thanks a lot 👍
I'm going to create a release now

**Comment by FlorianKempenich** [comment]
And it's in release: `2.6.0`

**Pull request merged**